### PR TITLE
Script to update aws-crt-java version number

### PR DESCRIPTION
**Pull request by graebm**
**ISSUE:**
The aws-crt-java dependency version number is hard-coded in several places. We kept forgetting to update it everywhere it needed to be updated.

**SOLUTION:**
Create script to update the version number everwhere that it's currently hardcoded. Run it like:
```sh
./update-crt.py 0.16.0
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by TwistedTwigleg — APPROVED** [review]
Looks good to me 👍

**Comment by TwistedTwigleg** [comment]
Minor question: Is there a way to automate this? I figure probably not, but thought I'd ask since it would remove the need to execute the script if it could be automated. To be fair though, I'm not sure it would be worth the time and energy to automate it though, since it might be messy trying to get that data from Git.

**Comment by graebm** [comment]
> Minor question: Is there a way to automate this? I figure probably not, but thought I'd ask since it would remove the need to execute the script if it could be automated. To be fair though, I'm not sure it would be worth the time and energy to automate it though, since it might be messy trying to get that data from Git.

We could make this more automated for sure, but it would be several days of work (some bot notices a new release of aws-crt-java, creates branch, runs this script to update dependencies, creates pull request, alerts team to merge if they like it). This stems the bleeding of a recurrent problem with 2hrs effort, and this script would be a part of that larger solution, so I'm going with this for now

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -2,9 +2,8 @@ name: CI
 
 on:
   push:
-    branches:
-      - '*'
-      - '!main'
```
Comment (graebm) [inline .github/workflows/ci.yml:7]: this change fixes the issue where CI doesn't run on branches with "/" in the name. Not sure how it fixes things, but it does

**Comment by TwistedTwigleg** [comment]
Totally makes sense! I was just curious. The script works for the current need and looks good, other stuff can always be added later if the need arises.

**Pull request merged**